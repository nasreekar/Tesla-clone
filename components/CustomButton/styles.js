import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
    buttonContainer: {
        width: '100%',
        padding: 8
    },
    buttonStyle: {
        height: 40,
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 20,
    },
    buttonText: {
        fontSize: 12,
        fontWeight: '500',
        textTransform: 'uppercase'
    }
});

export default styles;