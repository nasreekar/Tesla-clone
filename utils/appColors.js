export const appColors = {
    title: '#393c41',
    subtitle: '#898f9a',
    primaryButtonColor: '#171A20CC',
    primaryButtonTextColor: '#FFFFFF',
    secondaryButtonColor: '#FFFFFFA6',
    secondaryButtonTextColor: '#171A20'
}