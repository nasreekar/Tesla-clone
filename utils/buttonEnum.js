export const buttonEnum = { Primary: 'primary', Secondary: 'secondary' };