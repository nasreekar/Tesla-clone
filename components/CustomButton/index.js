import React from 'react'
import { Text, TouchableOpacity, View } from 'react-native'
import { buttonEnum } from '../../utils/buttonEnum'
import styles from './styles'

const CustomButton = (props) => {
    const { type, title, onPress } = props
    const buttonBackgroundColor = type === buttonEnum.Primary ? '#171A20CC' : '#FFFFFFA6'
    const buttonTextColor = type === buttonEnum.Primary ? '#FFFFFF' : '#171A20'

    return (
        <View style={styles.buttonContainer}>
            <TouchableOpacity
                style={[styles.buttonStyle, { backgroundColor: buttonBackgroundColor }]}
                onPress={onPress}>
                <Text style={[styles.buttonText, { color: buttonTextColor }]}>{title}</Text>
            </TouchableOpacity>
        </View>
    );
}

export default CustomButton;