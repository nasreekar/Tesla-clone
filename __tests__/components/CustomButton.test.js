import React from "react";
import { render } from "@testing-library/react-native";
import CustomButton from '../../components/CustomButton';
import { buttonEnum } from "../../utils/buttonEnum";
import { appColors } from "../../utils/appColors";

beforeEach(() => {
  jest.useFakeTimers();
});

describe("<CustomButton />", () => {

  const wrapper = render(<CustomButton type = {buttonEnum.Primary} title={"order now"} onPress ={()=>({})} />);
  const styles = wrapper.toJSON().props.style;
  const { backgroundColor } = styles;
  const childStyles = wrapper.toJSON().children[0].props.style;
  const {color: buttonTextColor} = childStyles;

  it("renders correctly", () => {
    const wrapper = render(<CustomButton type = {buttonEnum.Primary} title={"order now"} onPress ={()=>({})} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders primary button correctly",() => {
    const wrapper = render(<CustomButton type = {buttonEnum.Primary} title={"order now"} onPress ={()=>({})} />);
    const textComponent = wrapper.getByTestId('customButtonTitle');
    expect(textComponent.props.children).toEqual('order now');
  })

  it("renders secondary button correctly",() => {
    const wrapper = render(<CustomButton type = {buttonEnum.Secondary} title={"learn more"} onPress ={()=>({})} />);
    const textComponent = wrapper.getByTestId('customButtonTitle');
    expect(textComponent.props.children).toEqual('learn more');
  })

  it('primary button should be wrapper in black color', () => {
    const wrapper = render(<CustomButton type = {buttonEnum.Primary} title={"order now"} onPress ={()=>({})} />);
    const childStyles = wrapper.toJSON().children[0].props.style;
    const { backgroundColor } = childStyles;
    expect(backgroundColor).toEqual('#171A20CC');
  });

  it('secondary button should be wrapper in black color', () => {
    const wrapper = render(<CustomButton type = {buttonEnum.Secondary} title={"learn more"} onPress ={()=>({})} />);
    const childStyles = wrapper.toJSON().children[0].props.style;
    const { backgroundColor } = childStyles;
    expect(backgroundColor).toEqual('#FFFFFFA6');
  });
});
