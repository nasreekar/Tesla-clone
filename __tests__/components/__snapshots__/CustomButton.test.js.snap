// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 8,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#171A20CC",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "500",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
      testID="customButtonTitle"
    >
      order now
    </Text>
  </View>
</View>
`;
